index.js

```diff
@@ -43,7 +43,16 @@ module.exports = function (content) {
       typings += `}\ndeclare const styles: ${i};\nexport = styles;\n`;
     }
 
-    fs.writeFileSync(dtsPath, typings);
+    if (fs.existsSync(dtsPath)) {
+      const currentInput = fs.readFileSync(dtsPath, "utf-8");
+
+      // compare file contents ignoring whitespace
+      if (currentInput.replace(/\s+/g, "") !== typings.replace(/\s+/g, "")) {

```

Comment:
I think it would be enough to compare directly:
```js
if (!fs.existsSync(dtsPath) || fs.readFileSync(dtsPath, "utf-8") != typings) {
  fs.writeFileSync(dtsPath, typings, "utf8");
}
```